oonipipeline/src/oonipipeline/transforms/measurement_transformer.py

```diff
@@ -383,8 +400,17 @@ def measurement_to_tls_observation(
         tlso.ip = p.hostname
         tlso.port = p.port
 
-    tls_network_events = find_tls_handshake_network_events(tls_h, idx, network_events)
-    if tls_network_events:
+    tls_network_events: List[NetworkEvent] = None
+    if network_events:
+        # TODO(decfox): We check the first network event for the transaction_id and
+        # find tls handshake network events based on this. This is a weak check and
+        # somewhat sketchy 
+        if network_events[0] and network_events[0].transaction_id is None:
+            tls_network_events = find_tls_handshake_events_without_transaction_id(tls_h, idx, network_events)
```

Comment:
Maybe we can pass in here the measurement_uid so we can emit that in the warning messages so we have something to work with if we are to debug it.